include/nbl/builtin/hlsl/workgroup/broadcast.hlsl

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_WORKGROUP_BROADCAST_INCLUDED_
+#define _NBL_BUILTIN_HLSL_WORKGROUP_BROADCAST_INCLUDED_
+
+#include "nbl/builtin/hlsl/workgroup/ballot.hlsl"
+
+namespace nbl 
+{
+namespace hlsl
+{
+namespace workgroup
+{
+
+/**
+ * Broadcasts the value `val` of invocation index `id`
+ * to all other invocations.
+ * 
+ * We save the value in the shared array in the bitfieldDWORDs index 
+ * and then all invocations access that index.
+ */
+template<typename T, class ScratchAccessor>
+T Broadcast(in T val, in uint id)
+{
+// REVIEW: Check if we need edge barriers
+	ScratchAccessor scratch;
```

Comment:
call it `sharedAccessor` can't just be any scratch like SSBO....

or let the accessor define a `workgroupExecutionAndMemoryBarrier()` so then it will work with everything